python/snewpy/models/base.py

```diff
@@ -238,6 +238,9 @@ def get_initial_spectra(self, t, E, flavors=Flavor):
         initialspectra : dict
             Dictionary of model spectra, keyed by neutrino flavor.
         """
+        t = u.Quantity(t, ndmin=1)
+        E = u.Quantity(E, ndmin=2)
+
```

Comment:
I don't understand why both arrays don't have ndmin = 1 but for energy it is 2.

Reply:
You're right, it's not very clear here. 
1) I wanted to ensure that both `t` and `E` are converted to arrays even if they are only scalars
2) I wanted them to be 2D arrays, so everything relevant to `E` would be of shape `(len(E),1)` and `t` would be `(1, len(t))` - so after the calculations we'll have 2D array of shape `(len(E), len(t))`. **But** `t` will be used later in the `np.interp` method, which works only for 1D arrays. So I had to keep it 1D, and produce 1D arrays `L`,`Ea`,`a`, and later turn them to shape `[1, len(t)]` using `np.expand_dims`.

I will update the code to make it more clear.

Reply:
Please have a look now

Reply:
Okey, thanks for clarifying, I get now what you did, it makes sense to me